sensing/gnss_poser/src/gnss_poser_core.cpp

```diff
@@ -135,18 +139,23 @@ void GNSSPoser::callbackNavSatFix(
     canGetCovariance(*nav_sat_fix_msg_ptr) ? nav_sat_fix_msg_ptr->position_covariance[4] : 10.0;
   gnss_base_pose_cov_msg.pose.covariance[6 * 2 + 2] =
     canGetCovariance(*nav_sat_fix_msg_ptr) ? nav_sat_fix_msg_ptr->position_covariance[8] : 10.0;
-  gnss_base_pose_cov_msg.pose.covariance[6 * 3 + 3] = 0.1;
-  gnss_base_pose_cov_msg.pose.covariance[6 * 4 + 4] = 0.1;
-  gnss_base_pose_cov_msg.pose.covariance[6 * 5 + 5] = 1.0;
+  gnss_base_pose_cov_msg.pose.covariance[6 * 3 + 3] =
+    orientation_msg_ != nullptr ? std::pow(orientation_msg_->orientation.rmse_rotation_x, 2) : 0.1;
+  gnss_base_pose_cov_msg.pose.covariance[6 * 4 + 4] =
+    orientation_msg_ != nullptr ? std::pow(orientation_msg_->orientation.rmse_rotation_y, 2) : 0.1;
+  gnss_base_pose_cov_msg.pose.covariance[6 * 5 + 5] =
+    orientation_msg_ != nullptr ? std::pow(orientation_msg_->orientation.rmse_rotation_z, 2) : 1.0;
   pose_cov_pub_->publish(gnss_base_pose_cov_msg);
 
   // broadcast map to gnss_base_link
   publishTF(map_frame_, gnss_base_frame_, gnss_base_pose_msg);
 }
 
-void GNSSPoser::callbackNavPVT(const ublox_msgs::msg::NavPVT::ConstSharedPtr nav_pvt_msg_ptr)
+void GNSSPoser::callbackAutowareOrientation(
+  const autoware_sensing_msgs::msg::GnssInsOrientationStamped::ConstSharedPtr msg)
 {
-  nav_pvt_msg_ptr_ = nav_pvt_msg_ptr;
+  orientation_msg_->header = msg->header;
+  orientation_msg_->orientation = msg->orientation;
```

Comment:
```suggestion
  *orientation_msg_ = *msg;
```

You can replace it with this.

Reply:
Also better add a check like `if(orientation_msg_ != nullptr)`